### ISSUE TITLE
Security policy questions

### ISSUE THREAD

**Issue by theodoreb** [rollup/rollup#3980]
Documentation Is:

- [x] Missing
- [ ] Needed
- [ ] Confusing
- [ ] Not Sure?

### Please Explain in Detail...
The Drupal project is considering adding this library as one of our dependencies and so we're performing a standard stability review. We're looking into adopting this as a dev dependency of [@drupal/once](https://www.npmjs.com/package/@drupal/once) (corresponding Drupal core issue are [#2402103](https://www.drupal.org/project/drupal/issues/2402103) and [#3199444](https://www.drupal.org/project/once/issues/3199444)). I know rollup is not necessarily a runtime dependency and security issues might be less of a concern than some packages so I appreciate the time taken to consider this subject.

Since there isn't a policy at https://github.com/rollup/rollup/security I'm curious if you have any official policies documented somewhere regarding:

**Security releases**
For example, does more than one version receive security fixes, or only the current version? What would your policy on disclosure be? For example, would you ask users to report security issues privately, and publish the existence of the vulnerability only once a fix is available, for coordinated disclosure?
**Release windows/cadence**
For example, do they happen as necessary on any given day, or on a set schedule after a certain passage of time (e.g. once a month)? Looking at all the recent releases (which is great to see!), I can probably make some assumptions, but would like to confirm.
**Backwards compatibility guarantees**
rollup uses semver, so I assume the minor version promises not to break BC. Are there any guarantees that a given version will be supported for some period of time (an LTS version, for example)?

### Your Proposal for Changes
Add a few paragraph detailing the subjects above.

**Comment by theodoreb** [comment]
Thanks for the quick reply! I totally get that other projects have different ressources and priorities, the spirit of this issue is to make a few things more explicit, not ask for change on how things work in your project. For most JS packages having a release schedule is totally unreasonable, and it's not because we do things a certain way in Drupal that we expect others to follow, no pressure :)

In the spirit of adding clarity would it be possible to add some indication of what can people expect around version support, even saying "we only support the latest published version" would be helpful (and I'm sure others could be interested by this information). Same for the disclosure policy, having a couple of sentences that details what you have in mind by "do the right thing" is enough (for us at least).

If it helps, we asked similar questions for another library we're going to use: focus-trap/tabbable#136 and they ended up adding a short document which made explicit what they were already doing: https://github.com/focus-trap/tabbable/security/policy if that helps for inspiration.

Making what's happening currently explicit is already very helpful

**Comment by kzc** [comment]
> even saying "we only support the latest published version" would be helpful 

Like many other open source projects, this project has many contributors. It is "supported" in the sense that people try to make a best effort to improve the quality of the software as their interest and time allows. "Support" implies liability or a warranty. I'd refer you to the [license](https://github.com/rollup/rollup/blob/a12ec5b1125020c6509fb38a8b9218eea1301cc4/LICENSE.md) which states in part:

**THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.**

The Rollup distribution is built and bundled with some three dozen other project libraries, also listed in the [license](https://github.com/rollup/rollup/blob/a12ec5b1125020c6509fb38a8b9218eea1301cc4/LICENSE.md). Those projects could also be described as a "best effort", and have their own license terms.

**Comment by theodoreb** [comment]
I may not have been clear initially, I am not talking about a legally binding "support" or "warranty" in this case.

If I take Drupal as an example, all the code is GPL-v2 which has similar language around "warranty". Even then we do provide a [release schedule](https://www.drupal.org/about/core/policies/core-release-cycles/drupal-core-release-cycle-major-minor-and-patch-releases) and [security coverage informations](https://www.drupal.org/drupal-security-team/general-information#s-which-versions-are-supported). 

This is to show that a "no warranty" license doesn't automatically means that there is "no support of any kind", after all there is a discord channel to talk about rollup. Hence this issue to find out directly from the maintainers what is the reality of the situation.

**Comment by guybedford** [comment]
@theodoreb you're very much driving this here, so the point to bear in mind is that the maintainers of RollupJS all have other responsibilities and that this is very much an optional collaborative effort. If someone would like to fill this roll for the RollupJS project, we always welcome contributors, but short of someone volunteering or providing the resources it simply won't happen.

So without some kind of active participation the default of there being zero guarantees certainly holds. I know this may sound stand-offish, but you're coming from a project with a backing organization with almost $200m in funding and massive revenues - while we are grateful for the $20k per year we have to share with our one primary maintainer. Expecting us to take on more responsibilities for your company out of some arbitrary sense of duty under the moniker of open source is an attitude that must change for the health of the software ecosystem. Certainly let's produce quality software, but we should support it with proper foundations and not exploited labour.

**Comment by theodoreb** [comment]
Thank you for providing more informations around the organisation of the project. The response to these questions will inform how we introduce rollup in Drupal, and more importantly what kind of expectations people can have towards Drupal build tools and towards rollup itself. 

I understand that being more explicit than what is provided in the license is not a priority for you at this time. And on our side we'll make sure people don't rely on rollup for things the rollup community can't (or do not want to) provide at this time. We're happy to contribute to our upstream dependencies when necessary :)

The last thing I want is people being frustrated by mismatched expectations, from my experience frustration ends up making communication harder, which in turn increase the maintenance burden. I certainly don't wish that on anyone, that's where my need for clarity comes from and I appreciate the time taken to walk me through the though process behind the current policy.

@guybedford you mentioned a primary maintainer, am I correct in assuming he's aligned with what's been explained here?

**Comment by guybedford** [comment]
@theodoreb please don't interpret my response as a "no". I'm just clarifying that it won't get done if no one does it, and pushing back that I don't think it should be held against the project or its maintainers if none of us has the bandwidth to work on this.

I'd personally be open to the construction of a policy by a new or existing contributor to work on fleshing such a process out. Personally I do have some concerns with the ways in which security advisories create new burdens for open source maintainers, especially when the definition of an escalation is often not even very clearly defined. Such a process should start from threat models and clearly defining the types and response and what the boundaries of escalation are. We do have an open collective for the project where we take sponsorships - if you'd like to discuss this as part of a sponsorship offering we'd be more than happy to arrange a meeting further too.

We each speak for ourselves and no one single person is in charge. I can't speak for @lukastaegert and I don't know his opinion on the matter, but I certainly wouldn't want to make more work for him either.

**Comment by theodoreb** [comment]
We went another way ([simple copy/paste script](https://www.drupal.org/node/3197544)) and won't require rollup. Thanks you all for the answers so far.
